source/common/http/http1/codec_impl.cc

```diff
@@ -417,6 +419,13 @@ void ConnectionImpl::onHeaderValue(const char* data, size_t length) {
     return;
   }
 
+  if ((current_header_field_.size() + length + current_header_map_->byteSize()) >
+      (max_request_headers_kb_ * 1024)) {
```

Comment:
If this is checking onHeaderValue in the base class, isn't it now applying to response headers (when acting as an HTTP client) as well?

I think this either needs to be renamed to indicate that it affects both request and response headers (and the default passed in above may need to be adjusted to not have a functional change) or it should only apply to the request path.

Given we plan on doing this for response headers I'd suggest doing the former rather than the latter

Reply:
Yes, my concern is when we create 

std::make_unique<Http1::ClientConnectionImpl>(*connection_, *this, Http::DEFAULT_MAX_REQUEST_HEADERS_KB);

the client impl is sending request headers (no size checks on serialization) and receiving response headers (onHeaderValue expected to be called)

try cerr-ing in the http/1.1. integration test.  I'd expect the log to consist of the request headers received from the downstream client, and the response headers received from the upstream server.

Reply:
Gotcha, fixed this and added a test.